pkg/server/registry/apigroups/admin/computeclass/validator.go

```diff
@@ -39,6 +41,22 @@ func (s *ProjectValidator) Validate(ctx context.Context, obj runtime.Object) (re
 		}
 	}
 
+	invalidResources := []corev1.ResourceName{corev1.ResourceCPU, corev1.ResourceMemory}
+
+	for _, resource := range invalidResources {
+		if _, specified := cc.Resources.Requests[resource]; specified {
+			return append(
+				result,
+				field.Invalid(field.NewPath(fmt.Sprintf("spec.resources.requests%s", string(resource))), cc.Default, fmt.Sprintf("Cannot specifiy spec.resources.requests.%s. Use explicit spec.cpuScaler and spec.memory instead", string(resource))))
+		}
+	}
+
+	for _, resource := range invalidResources {
+		if _, specified := cc.Resources.Limits[resource]; specified {
+			return append(result, field.Invalid(field.NewPath(fmt.Sprintf("spec.resources.limits%s", string(resource))), cc.Default, fmt.Sprintf("Cannot specifiy spec.resources.limits.%s. Use explicit spec.cpuScaler and spec.memory instead", string(resource))))
+		}
+	}
```

Comment:
Any reason to not do this in one iteration?

```suggestion
	for _, resource := range invalidResources {
		if _, specified := cc.Resources.Requests[resource]; specified {
			return append(
				result,
				field.Invalid(field.NewPath(fmt.Sprintf("spec.resources.requests%s", string(resource))), cc.Default, fmt.Sprintf("Cannot specifiy spec.resources.requests.%s. Use explicit spec.cpuScaler and spec.memory instead", string(resource))))

		if _, specified := cc.Resources.Limits[resource]; specified {
			return append(result, field.Invalid(field.NewPath(fmt.Sprintf("spec.resources.limits%s", string(resource))), cc.Default, fmt.Sprintf("Cannot specifiy spec.resources.limits.%s. Use explicit spec.cpuScaler and spec.memory instead", string(resource))))
		}
	}
```

Reply:
Yeah, let's do this in one loop.

Reply:
absolutely! done